pcn-iptables: update and unify documentation

mbertrone opened the pull request:
- unify documentation under /Documentation
- remove docs under /src/components/iptables

My only doubt is about keeping or not the readme file under ` src/components/iptables/README.md`

frisso commented on Documentation/components/iptables/pcn-iptables.rst:
```diff
@@ -46,20 +46,34 @@ Limitations
+Install Steps
+^^^^^^^^^^^^^
+
 For ``pcn-iptables`` support you should enable ``ENABLE_PCN_IPTABLES`` flag in CMakeFile.
```
Does this mean that we must compile Polycube with a special flag?
If yes, I would rewrite this as follows:

To compile ``pcn-iptables``, you should enable `the `ENABLE_PCN_IPTABLES`` flag in the polycube CMakeFile; in fact, the set of eBPF services that are required by pcn-iptables to operate are not compiled with the by default:

mbertrone replied:
Just to be clear, the default config is:
`ENABLE_SERVICE_IPTABLES`=`ON`: this flag compiles and install the libpcn-iptables.so service (similar to others services like bridge, router, ...)
`ENABLE_PCN_IPTABLES` =`OFF`: this flag compile the customized version of iptables, and install the pcn-iptables-init pcn-iptables alias and pcn-iptables-clean scripts on your system.

frisso replied:
It was not clear to me, and what I understood was that libpcn-iptables.so was not created by default. I would suggest to change the text in order to make it more clear. Thanks!

mbertrone replied:
Agree. I've added a more detailed description about flags.

frisso commented on Documentation/components/iptables/pcn-iptables.rst:
```diff
@@ -129,4 +143,20 @@ Limitations
 - It requires your network interfaces to support XDP Native mode
-- If any interface is not supporting XDP, on such interface traffic is not filtered
\ No newline at end of file
+- If any interface is not supporting XDP, on such interface traffic is not filtered
```
Should we write this as follows?
pcn-iptables operates only on interfaces that support XDP (native or generic?) mode; traffic is not filtered on interfaces that support only eBPF TC programs.

frisso commented on Documentation/components/iptables/pcn-iptables.rst:
```diff
@@ -129,4 +143,20 @@ Limitations
+iptables submodule
+^^^^^^^^^^^^^^^^^^
+
+Under :scm_web:`src/components/iptables/iptables <src/components/iptables>` a customized fork of iptables is included as submodule.
```
Turn the sentence the other way around.
A customized fork of iptables is included as submodule under...

frisso commented on Documentation/components/iptables/pcn-iptables.rst:
```diff
@@ -129,4 +143,20 @@ Limitations
+^^^^^^^^^^^^^^^^^^
+
+Under :scm_web:`src/components/iptables/iptables <src/components/iptables>` a customized fork of iptables is included as submodule.
+We customized this version of iptables in order not to inject iptables command into netfilter, but convert them, after a validation, into polycube syntax.
```
validation
==>
validation pass (or validation step)

frisso commented:
@mbertrone: My only doubt is about keeping or not the readme file under src/components/iptables/README.md
==> I would remove its content and simply point to the online documentation.

mbertrone commented:
> @mbertrone: My only doubt is about keeping or not the readme file under src/components/iptables/README.md
> ==> I would remove its content and simply point to the online documentation.

@frisso I've added a link to online documentation and addressed all previous comments.
Thanks for the review

frisso commented:
@acloudiator Feel free to merge when you are also ok.

frisso commented:
@acloudiator Is there any problem that you added the "hold" label? Please let us know what feels is missing.

acloudiator commented:
> @acloudiator Is there any problem that you added the "hold" label? Please let us know what feels is missing.

This is about the documentation that we're in a discussion.

mbertrone commented:
@acloudiator @frisso I updated the link in README file. Now it points to .rst file inside /Documentation folder.

frisso commented:
@acloudiator LGTM. Can we merge?
